java/java.hints/src/org/netbeans/modules/java/hints/jdk/CheckRegexTopComponent.java

```diff
@@ -0,0 +1,249 @@
+        matchPattern();
+    }//GEN-LAST:event_RegexTextAreaKeyReleased
+
+    // Variables declaration - do not modify//GEN-BEGIN:variables
+    private javax.swing.JLabel ExampleLabel;
+    private javax.swing.JScrollPane ExampleScrollPane;
+    private javax.swing.JTextArea ExampleTextArea;
+    private javax.swing.JLabel RegexLabel;
+    private javax.swing.JScrollPane RegexScrollPane;
+    private javax.swing.JTextArea RegexTextArea;
+    // End of variables declaration//GEN-END:variables
+    @Override
+    public void componentOpened() {
+        // TODO add custom code on component opening
+    }
+
+    @Override
+    public void componentClosed() {
+        // TODO add custom code on component closing
+    }
+
+    void writeProperties(java.util.Properties p) {
+        // better to version settings since initial version as advocated at
+        // http://wiki.apidesign.org/wiki/PropertyFiles
+        p.setProperty("version", "1.0");    // NOI18N
+        // TODO store your settings
+    }
+
+    void readProperties(java.util.Properties p) {
+        String version = p.getProperty("version");  // NOI18N
+        // TODO read your settings according to their version
+    }
+
+    public static synchronized CheckRegexTopComponent getDefault() {
+        if (instance == null) {
+            instance = new CheckRegexTopComponent();
+        }
+        Mode outputMode = WindowManager.getDefault().findMode("output");
+        if (outputMode != null) {
+            outputMode.dockInto(instance);
+        }
+        return instance;
+    }
+
+    public static synchronized CheckRegexTopComponent findInstance() {
+        TopComponent win = WindowManager.getDefault().findTopComponent("CheckRegexTopComponent");
+        if (win == null) {
+            Logger.getLogger(CheckRegexTopComponent.class.getName()).warning(
+                    "Cannot find " + "CheckRegexTopComponent" + " component. It will not be located properly in the window system.");
+            return getDefault();
+        }
+        if (win instanceof CheckRegexTopComponent) {
+            return (CheckRegexTopComponent) win;
+        }
+        Logger.getLogger(CheckRegexTopComponent.class.getName()).warning(
+                "There seem to be multiple components with the '" + "CheckRegexTopComponent"
+                + "' ID. That is a potential source of errors and unexpected behavior.");
+        return getDefault();
+    }
+
+    void setData(String origString) {
+        RegexTextArea.setText(origString);
+    }
+
+    private void matchPattern() {
+        Pattern p = Pattern.compile(RegexTextArea.getText().trim());
+        Matcher m = p.matcher(ExampleTextArea.getText().trim());
+
+        Highlighter highlighter = ExampleTextArea.getHighlighter();
+        highlighter.removeAllHighlights();
+
+        if (m.matches()) {
+            ExampleTextArea.setForeground(Color.GREEN);
+            ExampleTextArea.setToolTipText("The Example Matches the Regular Expression");
```

Comment:
Maybe, I think users expect this to be translated.
For such that cases, please use `@NbBundle.Messages`.
e.g. 

```java
@NbBundle.Messages({
    "CheckRegexTopComponent.tooltip.match.regex=The Example Matches the Regular Expression",
    "CheckRegexTopComponent.tooltip.need.more.input=Need more input to match",
    ...
})
private void matchPattern() {

...
            exampleTextArea.setToolTipText(Bundle.CheckRegexTopComponent_tooltip_match_regex());
```